### PR TITLE
[Alter]Clear expire alterJobV2

### PR DESCRIPTION
**Pull request by WingsGo**
Too much AlterJobsV2 may consume too much memory, which may cause FullGC. Clear some data for finished or cancelled alterJobs and remove them when expire.

### CODE REVIEW

**Comment by WingsGo** [comment]
For #3132

File: fe/src/main/java/org/apache/doris/alter/AlterHandler.java [hunk 1]
```diff
@@ -119,6 +121,30 @@ public AlterJobV2 getUnfinishedAlterJobV2ByJobId(long jobId) {
...
+                        historyJob.getTableId(), TimeUtils.longToTimeString(historyJob.getCreateTimeMs()));
+            }
+        }
+    }
+
+    private void clearExpireFinishedOrCancelledAlterJobsV2() {
+        for (AlterJobV2 alterJobV2 : finishedOrCancelledAlterJobsV2) {
+            if (alterJobV2.isExpire()) {
+                finishedOrCancelledAlterJobsV2.remove(alterJobV2);
+                LOG.info("remove history {} jobV2[{}]. created at {}", alterJobV2.getType(),
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/alter/AlterHandler.java]:
```suggestion
                LOG.info("remove history {} jobV2[{}]. finish at {}", alterJobV2.getType(),
```

File: fe/src/main/java/org/apache/doris/alter/AlterHandler.java [hunk 2]
```diff
@@ -119,6 +121,30 @@ public AlterJobV2 getUnfinishedAlterJobV2ByJobId(long jobId) {
...
 
+    private void clearExpireFinishedOrCancelledAlterJobs() {
+        long curTime = System.currentTimeMillis();
+        // clean history job
+        Iterator<AlterJob> iter = finishedOrCancelledAlterJobs.iterator();
+        while (iter.hasNext()) {
+            AlterJob historyJob = iter.next();
+            if ((curTime - historyJob.getCreateTimeMs()) / 1000 > Config.history_job_keep_max_second) {
+                iter.remove();
+                LOG.info("remove history {} job[{}]. created at {}", historyJob.getType(),
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/alter/AlterHandler.java]: change to "finish at"?

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**